packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -556,6 +569,13 @@ export class FluidDataStoreRuntime extends EventEmitter implements IFluidDataSto
                 return new TreeTreeEntry(key, snapshot);
             }));
 
+        // NOTE: Search blob concept!
+        // If there is an extra contract to be executed, then do so and push the results onto the tree:
+        if (this.extraSnapshotContract !== undefined) {
+            const extras = await this.extraSnapshotContract();
```

Comment:
It's probable fine for now to have just one callback, but as we discussed in another PR, to support synchronous and async workflows, we likely need multiple callbacks, and thus system likely needs plural here